Remove duplicates before "inversions" are inverted

Wansmer opened the pull request:
First, thanks for a great plugin.

In this PR
1. Implemented the function to remove duplicates before "inversions" are inverted.
Motivation: I think it's not exactly an "error" to disable plugin functionality when user add duplicate rules. (I did it myself when adding `['right] = 'left'` and had no idea why the plugin didn't work);

2. Replaced `print()` to `vim.notify()` to display an error, because if it had been added earlier, I would have found my error at the same time.

Hope it will be helpful.

Wansmer committed:
implemented remove duplicates function

Wansmer committed:
replace print to vim.notify with warn level

nguyenvukhang commented:
`utils.remove_duplicates` should return a sanitized list, but if the input has `["a"] = "c"` and `["b"] = "c"`, the following reserve lookup call can still fail.

Nice spot on the `vim.notify` though. If you don't want to fix the bug above I'll merge it for this change.

Wansmer committed:
improved sanitizing list

Wansmer commented:
@nguyenvukhang, I fixed it at [c5b462e](https://github.com/nguyenvukhang/nvim-toggler/pull/8/commits/c5b462e0413a2bd9889d015fb83706af7b7b4c21):
1. Renamed function `remover_duplicates` to `sanitize_list` for better semantic;
2. Added additional rules for validation pairs key/value;

Note: key/value like ["a"] = "c" and ["b"] = "c" includes once but no guarantees order (it may be  ["a"] = "c" or ["b"] = "c" in resulted list). I guess it because before called `table_extend`.

nguyenvukhang commented:
btw there was one last case this didn't account for, and that is 
`{ ['a'] = 'b', ['b'] = 'c' }`

(now accounted for and tested on 9d8f81c)

nguyenvukhang commented:
It'd be cool if future PRs (not just for this project) came with tests.

Either way, thanks for contributing 👍🏽

Wansmer commented:
Cool refactoring! Btw, I tried to run `NvimTogglerTest` and get some errors. Fixed by PR #9.
